Modules/_sqlite/connection.c

```diff
@@ -1998,10 +2003,28 @@ pysqlite_connection_iterdump_impl(pysqlite_Connection *self)
         }
         return NULL;
     }
+    PyObject *args[2] = {(PyObject *)self, filter};
+    PyObject *kwnames = PyTuple_New(1);
+    PyObject *py_filter = NULL;
+
+    if (!kwnames) {
+        goto error;
+    }
+    py_filter = PyUnicode_FromString("filter");
+    if (!py_filter) {
+        goto error;
+    }
+    PyTuple_SET_ITEM(kwnames, 0, py_filter);
 
-    PyObject *retval = PyObject_CallOneArg(iterdump, (PyObject *)self);
+    PyObject *retval = PyObject_Vectorcall(iterdump, args, 1, kwnames);
```

Comment:
We also need to decref `kwnames` and `py_filter` before returning on the "happy path".

Reply:
Ah, yes, `PyTuple_SET_ITEM` is one of the weird ones; good call! :)